omnigibson/systems/micro_particle_system.py

```diff
@@ -904,8 +906,9 @@ def generate_particles_from_link(
         # Check which points are inside the volume and only keep those
         particle_positions = particle_positions[np.where(check_in_volume(particle_positions))[0]]
 
-        # Also prune any that in contact with anything
-        particle_positions = particle_positions[np.where(cls.check_in_contact(particle_positions))[0]]
+        # Also prune any that in contact with anything if requested
+        if check_contact:
+            particle_positions = particle_positions[np.where(cls.check_in_contact(particle_positions) == 0)[0]]
```

Comment:
why do we optionally `check_in_contact`? Is it for performance reasons?

Reply:
Actually, 2 reasons:
1. Performance, as you mentioned
2. To avoid inaccurate results. Case in point: Dicing: where we teleport / destroy the object, but physics doesn't get updated until you take a step, so the check_in_contact call will actually be invalid. So we don't run check_in_contact in that case